pkg/pmem-csi-operator/controller/deployment/deployment_controller.go

```diff
@@ -193,24 +193,24 @@ func (r *ReconcileDeployment) Get(obj runtime.Object) error {
 	return r.client.Get(context.TODO(), key, obj)
 }
 
-// Create create new Kubernetes object
-func (r *ReconcileDeployment) Create(obj runtime.Object) error {
-	err := r.Get(obj)
+// CreateOrUpdate tries to get if any object with existing creates new Kubernetes object if not existing else
+func (r *ReconcileDeployment) CreateOrUpdate(obj runtime.Object) error {
+	objCopy := obj.DeepCopyObject()
+	metaObj, _ := meta.Accessor(obj)
+	err := r.Get(objCopy)
 	if err == nil {
-		// Already found an active object
-		return nil
+		// FIXME(avalluri): Check the ownership of the object(?) to make sure
+		// that was created by the same deployment
+		klog.Infof("Updating: %q of type %q ", metaObj.GetName(), obj.GetObjectKind().GroupVersionKind())
+		// Update existing active object,
+		return r.client.Update(context.TODO(), obj)
```

Comment:
I doubt that the apiserver will accept the update because the resourceVersion won't match.

Write an E2E test for this and we'll know...

Reply:
Yes, true. I fixed this by choosing relevant metadata from the object returned by the API server to our new object before update.